mmgen/models/architectures/sngan_proj/generator_discriminator.py

```diff
@@ -2,100 +2,86 @@
-        # if isinstance(output_scale, int):
-        #     pass
-
         self.input_scale = input_scale
         self.output_scale = output_scale
         self.noise_size = noise_size
 
-        self.res_block_cfg = deepcopy(self._default_res_block_cfg)
-        if res_block_cfg is not None:
-            self.res_block_cfg.update(res_block_cfg)
-        if 'auto_sync_bn' not in self.res_block_cfg:
-            self.res_block_cfg['auto_sync_bn'] = auto_sync_bn
-        self.res_block_cfg['num_classes'] = num_classes
+        self.blocks_cfg = deepcopy(self._default_blocks_cfg)
+        if blocks_cfg is not None:
+            self.blocks_cfg.update(blocks_cfg)
+        if 'auto_sync_bn' not in self.blocks_cfg:
+            self.blocks_cfg['auto_sync_bn'] = auto_sync_bn
+        self.blocks_cfg['num_classes'] = num_classes
+        self.blocks_cfg['act_cfg'] = act_cfg
 
         channel_cfg = self._defualt_channels_cfg if channel_cfg is None \
             else channel_cfg
         if isinstance(channel_cfg, dict):
             if output_scale not in channel_cfg:
-                raise KeyError(
-                    '`output_scale={} is not found in `channel_cfg`, '
-                    'only support configs for {}'.format(
-                        output_scale, [chn for chn in channel_cfg.keys()]))
+                raise KeyError(f'`output_scale={output_scale} is not found in '
+                               '`channel_cfg`, only support configs for '
+                               f'{[chn for chn in channel_cfg.keys()]}')
             self.channel_factor_list = channel_cfg[output_scale]
         elif isinstance(channel_cfg, list):
             self.channel_factor_list = channel_cfg
         else:
             raise ValueError('Only support list or dict for `channel_cfg`, '
-                             'receive {}'.format(type(channel_cfg)))
+                             f'receive {type(channel_cfg)}')
 
         self.noise2feat = nn.Linear(
             noise_size,
             input_scale**2 * base_channels * self.channel_factor_list[0])
-        self.res_blocks = nn.ModuleList()
+
+        self.conv_blocks = nn.ModuleList()
         for idx in range(len(self.channel_factor_list)):
             factor_input = self.channel_factor_list[idx]
             factor_output = self.channel_factor_list[idx+1] \
                 if idx < len(self.channel_factor_list)-1 else 1
 
-            self.res_blocks.append(
-                SNGAN_GenResBlock(factor_input * base_channels,
-                                  factor_output * base_channels,
-                                  **self.res_block_cfg))
+            # get block-specific config
+            block_cfg_ = deepcopy(self.blocks_cfg)
+            block_cfg_['in_channels'] = factor_input * base_channels
+            block_cfg_['out_channels'] = factor_output * base_channels
+            self.conv_blocks.append(build_from_cfg(block_cfg_, MODULES))
```

Comment:
In `MMGeneration`, you can directly adopt `build_module` in this way:

```python
from mmgen.model import build_module

m = build_module(config)
```

Reply:
Fixed this. `default_args` has also add to `build_module()`

Reply:
However, it seems that you have not switched to `build_module` function.